src/export/ExportFilePanel.cpp

```diff
@@ -371,23 +372,36 @@ void ExportFilePanel::ValidateAndFixExt()
 
    wxFileName filename;
    filename.SetFullName(mFullName->GetValue());
+   const auto desiredExt = filename.GetExt().Trim();
 
-   //Remove extra whitespaces
-   auto desiredExt = filename.GetExt().Trim().Trim(false);
-
-   auto it = std::find_if(
-      formatInfo.extensions.begin(),
-      formatInfo.extensions.end(),
-      // if typed extension uses different case (e.g. MP3 instead of mp3)
-      // we'll reset the file extension to one provided by FormatInfo
-      [&](const auto& ext) { return desiredExt.IsSameAs(ext, false); });
+   //See https://github.com/audacity/audacity/issues/5823
+   //check if extension is valid, i.e. does not contain whitespace characters.
+   //Otherwise everything after '.'(if present) is considered to be a part of name.
+   if(wxRegEx{R"(^[^ ]+$)"}.Matches(desiredExt))
+   {
+      auto it = std::find_if(
```

Comment:
There remains a problem, at least on Windows: with "Artist – Track (feat.Artist)" (no whitespace before "Artist)"), "Artist)" is still going to be considered an extension and this branch will get executed, resulting in e.g. "Artist – Track (feat.wav".
I think there should be a proper filename extension regex here. Would `R"^([a-zA-Z0-9]+)$" work?

Reply:
If you give that name to a file on Windows it will perfectly accept ".Artist)" as file extension, though unknown may be.